src/viam/examples/modules/simple/CMakeLists.txt

```diff
@@ -25,3 +25,18 @@ install(
   TARGETS simple_module
   COMPONENT examples
 )
+
+add_executable(simple_module_client)
+target_sources(simple_module_client
```

Comment:
just for my own curiosity is there a reason we do it this way (also for `add_library` calls in the sdk) instead of just `add_executable(simple_module_client client.cpp)`

Reply:
That's my doing. I like being consistent about using the `target_xyz` calls to affect the target rather than having some of them apply at the declaration and others (e.g. `target_link_libraries`) be called for side effects. Since `target_sources` works like `target_link_libraries` I used it consistently everywhere, and left the `add_executable` and `add_library` calls are purely introducing the name with no side effects.

Reply:
cool that makes sense!